[SMDDP] RNNT Notebook

Zha0q1 opened the pull request:
This pr adds a new notebook for RNNT, which is a deep learning example forked from https://github.com/mlcommons/training_results_v1.0. The model has been adapted to work with SMDDP and hosted at https://github.com/HerringForks/SMDDP-Examples/tree/main/pytorch/rnnt

Zha0q1 commented:
hi @ajaykarpur @hsl89 would you suggest the next steps to get this pr reviewed and merged? thanks!

hsl89 commented:
@jkroll-aws , this notebook has the same Fsx issue we discussed before. what's the latest guidelines for it? seems that CI is skipping this notebook

Zha0q1 commented:
> @jkroll-aws , this notebook has the same Fsx issue we discussed before. what's the latest guidelines for it? seems that CI is skipping this notebook

Thanks for chiming in! Can I ask what's the fsx issue?

jkroll-aws commented on training/distributed_training/pytorch/data_parallel/rnnt/RNNT_notebook.ipynb:
```diff
@@ -0,0 +1,371 @@
+{
```
<p>Typos: <code>prarameters</code> ﻿, <code>traning</code>  <span class="ql-cursor">﻿</span></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/2875/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

mchoi8739 replied:
<p>LibriSpeech dataset seems not approved by Legal for public commercial use. I'll sync up with you offline.</p>

mchoi8739 replied:
<p>Submitted a ticket for a legal approval </p>

Zha0q1 replied:
ticket is approved! I added a link to the license in the notebook

jkroll-aws commented on training/distributed_training/pytorch/data_parallel/rnnt/RNNT_notebook.ipynb:
```diff
@@ -0,0 +1,371 @@
+{
```
<p>Grammar: <code>SageMaker by default use</code>  -&gt; <code>uses</code>  <span class="ql-cursor">﻿</span></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/2875/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

jkroll-aws replied:
<p>Can you link to the exact file of the mlcommons example that was adapted for this notebook?</p>

Zha0q1 replied:
added thanks!

jkroll-aws commented on training/distributed_training/pytorch/data_parallel/rnnt/RNNT_notebook.ipynb:
```diff
@@ -0,0 +1,371 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #3.      aws ecr get-login-password --region us-east-1 | docker login \
</pre><p>Don't hard-code the region.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/2875/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Zha0q1 replied:
fixed!

jkroll-aws commented on training/distributed_training/pytorch/data_parallel/rnnt/RNNT_notebook.ipynb:
```diff
@@ -0,0 +1,371 @@
+{
```
<p>It's cleaner to use hyperlinks rather than raw links - use the doc page as the hyperlink text.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/2875/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Zha0q1 replied:
<p>fixed thanks!</p>

jkroll-aws commented on training/distributed_training/pytorch/data_parallel/rnnt/RNNT_notebook.ipynb:
```diff
@@ -0,0 +1,371 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #4.    region = "us-west-2"  # Example: us-west-2
</pre><p>Set region to be whatever the current default region is for the customer.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/2875/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Zha0q1 replied:
fixed!

jkroll-aws commented:
Please install `pip install black-nb` and run black-nb -l 100 {notebook-name}.ipynb` in order to reformat the notebook and pass the code formatting check.

jkroll-aws commented:
Please review the output from the sagemaker-examples-grammar check and fix relevant spelling/grammar errors (you can skip anything related to SageMaker/ML terminology). To see the results, click on `Build Logs` in the box containing the check.

jkroll-aws commented on training/distributed_training/pytorch/data_parallel/rnnt/RNNT_notebook.ipynb:
```diff
@@ -0,0 +1,371 @@
+{
```
<ol><li>For step 1, <code>Download and prepare your training dataset on S3</code>	, are you expecting the customer to download LibriSpeech? You mention and link it earlier but don't provide any more info on dataset usage or details.</li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/2875/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Zha0q1 replied:
<p>fixed!</p>

hsl89 commented:
@Zha0q1 , when using fsx, users need to manually input their security group id, file system id, etc, this means the notebook cannot be automatically tested by using tools like papermill

jkroll-aws commented:
> @jkroll-aws , this notebook has the same Fsx issue we discussed before. what's the latest guidelines for it? seems that CI is skipping this notebook

I added notebooks that use FSx to be auto-skipped by CI until we have a fix for being able to test them.

Zha0q1 commented:
hi @jkroll-aws, I have addressed the comments, fixed the spelling and run black-nb to reformat the notebook. Would you help review again and suggest any further action items? thanks!

Zha0q1 commented:
hi @jkroll-aws, I have addressed the comments, fixed the spelling and run black-nb to reformat the notebook. Would you help review again and suggest any further action items? thanks!
